### PR TITLE
Review of user manual (only typos and syntax)

### PR DESCRIPTION
**Pull request by hexcoder-**
I reviewed the user manual up to before section 10.7 (part 1) with regard to typos and consistency. Since I am neither a native speaker nor an expert, please review my changes, thanks.

### CODE REVIEW

File: manual/parts/threads.tex [hunk 1]
```diff
@@ -123,10 +123,10 @@ \subsection{Scheduling Contexts}
 represent a lower bound on execution, as a thread must have the highest or equal highest priority
 of all runnable threads to be guaranteed to be scheduled at all, and the kernel does not conduct
 an admission test. As a result the set of all parameters is not necessarily schedulable. If
-multiple threads have available budget concurrently they are scheduled first-in first-out, and
+multiple threads have budgets available concurrently they are scheduled first-in first-out, and
 round-robin scheduling is applied once the budget is expired.
 
-A scheduling context that is eligible to be picked by the scheduler, i.e has budget available, is
+A scheduling context that is eligible to be picked by the scheduler, i.e has a budget available, is
```
Comment (lsf37) [inline manual/parts/threads.tex:129]: I'd leave this one without article.

```suggestion
A scheduling context that is eligible to be picked by the scheduler, i.e has budget available, is
```

File: manual/parts/threads.tex [hunk 2]
```diff
@@ -169,15 +169,15 @@ \subsection{Scheduling Contexts}
 
 Threads that have short execution times (e.g interrupt handlers) and are not frequently preempted
 should have less refills, while longer running threads with long values of $b$ should have a higher
-value. Threads bound to a scheduling context with 0 extra refills will behave periodically -- tasks
+value. Threads bound to a scheduling context with 0 extra refills will get run periodically -- tasks
```
Comment (lsf37) [inline manual/parts/threads.tex:172]:
```suggestion
value. Threads bound to a scheduling context with 0 extra refills will run periodically -- tasks
```

File: manual/parts/threads.tex [hunk 3]
```diff
@@ -169,15 +169,15 @@ \subsection{Scheduling Contexts}
...
 that use their head replenishment, or call yield, will not be scheduled again until the start of
 their next period.
 
 Given the number of replenishments is limited, if a node's SC changes and the outgoing SC does not
 have enough space to store the new replenishment, space is created by removing the current
 replenishment which can result in preemption if the next replenishment is not yet available.
 Scheduling contexts with a higher number of configured refills will consume closer
-to their whole budget, as they can be preempted or switch threads more often without filling their
+to their whole budget, as they can be preempted or else? switch threads more often without filling their
```
Comment (lsf37) [inline manual/parts/threads.tex:180]: as a logician I'm tempted to leave the original `or`, but I think language pragmatics probably want `and` here:

```suggestion
to their whole budget, as they can be preempted and switch threads more often without filling their
```

File: manual/parts/threads.tex [hunk 4]
```diff
@@ -211,7 +211,7 @@ \subsection{Scheduling Context Creation} \label{sec:sc_creation}
 time.  Threads run on the node that their scheduling context is configured for.  Scheduling context
 parameters can then be set and updated using
 \apifunc{seL4\_SchedControl\_ConfigureFlags}{schedcontrol_configureflags}, which allows the budget and period
-to be specified along with a bitwise OR'd set of the following flags.
+to be specified along with a bitwise OR'd set of the following flags:
```
Comment (lsf37) [inline manual/parts/threads.tex:214]: I don't know how consistent we are in the manual with that, but journal styles would usually not put a colon here, but a full stop as in the original.

```suggestion
to be specified along with a bitwise OR'd set of the following flags.
```

File: manual/parts/threads.tex [hunk 5]
```diff
@@ -553,18 +553,22 @@ \subsection{Debug Exception: Single-stepping}
 is currently configured (if any) for single-stepping will be the implicit
 \texttt{bp\_num} argument in a single-step debug fault reply.
 
-The kernel's single-stepping, also supports skipping a certain number of
+% The description for skipping num_instructions had an off-by-one error
+% I.e. num_instruction = 1 does not skip any instructions between stopps
+% num_instruction = 2 "skips" 1 instruction, but executes 2 instructions before stopping
```
Comment (lsf37) [inline manual/parts/threads.tex:558]: Thanks for fixing that.

File: manual/parts/threads.tex [hunk 6]
```diff
@@ -594,6 +598,8 @@ \subsection{Timeout Fault (MCS only)}
     \\ \bottomrule \end{tabularx}\\ \\ \caption{\label{tbl:tf_message_32}
     Timeout fault outcome on 32-bit architectures.} \end{table}
 
+% above table might be incomplete. What is the outcome for 64-bit architectures?
```
Comment (lsf37) [inline manual/parts/threads.tex:601]: @indan, should that be TODO/FIXME?

**Review by lsf37 — APPROVED** [review]
Thanks for that, this is great!

@Indanz, @bbrcknl does this conflict with other PR's you have open on updating the manual?

**Comment by lsf37** [comment]
The commits should probably be squashed into a single commit when we merge, no need for one per file.

**Comment by hexcoder-** [comment]
Regarding a squashed merge, I agree. Will have to investigate that first.

File: manual/parts/cspace.tex [hunk 7]
```diff
@@ -65,7 +65,7 @@ \subsection{CSpace Creation}
 Like any other object, a \obj{CNode} must be created by calling
 \apifunc{seL4\_Untyped\_Retype}{untyped_retype} on an appropriate
 amount of untyped memory (see \autoref{sec:object_sizes}).  The caller
-must therefore have a capability to enough untyped memory as well as
+must therefore have a capability to ensure enough untyped memory as well as
```
Comment (axel-h) [inline manual/parts/cspace.tex:68]: I think "ensure" this should not be added here, or there sentence needs more rewording. You need *one* untyped mem cap that is at least the size that is needed for the c-space

File: manual/parts/objects.tex [hunk 8]
```diff
@@ -44,8 +44,8 @@ \chapter{\label{ch:objects}Kernel Services and Objects}
     kernel services along with their book-keeping information.
 \end{description}
 
-This chapter gives an overview of these services, describes how kernel
-objects are accessed by userspace applications, and describes how new
+This chapter gives an overview of these services and describes how kernel
```
Comment (axel-h) [inline manual/parts/objects.tex:47]: leave the comma here?

File: manual/parts/objects.tex [hunk 9]
```diff
@@ -173,7 +173,7 @@ \section{System Calls}
     section \ref{sec:faults}) when attempted with other capability types.
 
     In the MCS configuration, \emph{Receive} takes a reply capability---a
-    capability to a reply object--as a parameter.
+    capability to a reply object---as a parameter.
```
Comment (axel-h) [inline manual/parts/objects.tex:176]: should also put a space before and after `---` then?

Comment (lsf37) [reply manual/parts/objects.tex:176]: depends on the style, the manual doesn't put spaces there -- you'd have either have to update all of them or none of them. I think the change is correct as is.

File: manual/parts/objects.tex (hunk 8 above)
Comment (lsf37) [reply manual/parts/objects.tex:47]: I think the change is correct, there should not be a comma here in English (actually in German neither ;-))

**Comment by lsf37** [comment]
> Regarding a squashed merge, I agree. Will have to investigate that first.

All good, you don't have to do anything for that, there is a "Squash and merge" option we can use on GitHub for it.

**Comment by hexcoder-** [comment]
I finished the review now (see Review part2).

File: manual/parts/ipc.tex [hunk 10]
```diff
@@ -207,7 +207,7 @@ \subsection{Errors}
 
     \item The destination slot cannot be looked up. Unlike the send
     system call, the \apifunc{seL4\_Recv}{sel4_recv} system call does not check that the
-    destination slot exists and is empty before it initiates the receive.
```
Comment (bbrcknl) [inline manual/parts/ipc.tex:210]: Should this actually be left as ```receive```? I'm not sure ```reception``` is the correct technical term, but I'm not an expert :-)

**Review by hexcoder- — COMMENTED** [review]
I am done so far, waiting for comments from reviewer

File: manual/parts/ipc.tex [hunk 11]
```diff
@@ -207,7 +207,7 @@ \subsection{Errors}
...
+    destination slot exists and is empty before it initiates the delivery.
```
Comment (lsf37) [inline manual/parts/ipc.tex:210]:
```suggestion
    destination slot exists and is empty before it initiates the receive operation.
```

File: libsel4/arch_include/arm/interfaces/sel4arch.xml [hunk 12]
```diff
@@ -15,7 +15,7 @@
             <description>
                 Takes a VSpace capability as an argument,
                 and installs a reference to the invoked
-                <texttt text='PageTable'/> in the VSpace according to the provided virtual address.
+                <texttt text='Page Table'/> in the VSpace according to the provided virtual address.
```
Comment (lsf37) [inline libsel4/arch_include/arm/interfaces/sel4arch.xml:18]:
```suggestion
                <texttt text='PageTableCap'/> in the VSpace according to the provided virtual address.
```

Comment (lsf37) [inline libsel4/arch_include/arm/interfaces/sel4arch.xml:18]: Needs slightly more reformulation. What is being invoked is a PageTableCap, but the reference being installed is to a page table (the one the cap points to)

File: manual/parts/cspace.tex (hunk 7 above)
Comment (lsf37) [reply manual/parts/cspace.tex]: You’re right, ensure is not the right word. Maybe “a capability to sufficiently large untyped memory as well as..”

**Review by lsf37 — APPROVED** [review]
Have tweaked the two remaining review feedback comments and squashed all commits into one. This is now ready to merge from my side.

**Pull request merged**